plugins/modules/docker_compose_v2.py

```diff
@@ -0,0 +1,923 @@
+      - Equivalent to C(docker-compose --project-name).
+    type: str
+  env_file:
+    description:
+      - By default environment files are loaded from a C(.env) file located directly under the I(project_src) directory.
+      - I(env_file) can be used to specify the path of a custom environment file instead.
+      - The path is relative to the I(project_src) directory.
+      - Equivalent to C(docker-compose --env-file).
+    type: path
+  files:
+    description:
+      - List of Compose files.
+      - Files are passed to docker-compose in the order given.
+      - Equivalent to C(docker-compose -f).
+    type: list
+    elements: path
+  profiles:
+    description:
+      - List of profiles to enable when starting services.
+      - Equivalent to C(docker-compose --profile).
+    type: list
+    elements: str
+  state:
+    description:
+      - Desired state of the project.
+      - Specifying C(pulled) is the same as running C(docker-compose pull).
+      - Specifying C(built) is the same as running C(docker-compose build).
+      - Specifying C(stopped) is the same as running C(docker-compose stop).
+      - Specifying C(present) is the same as running C(docker-compose up).
+      - Specifying C(restarted) is the same as running C(docker-compose restart).
+      - Specifying C(absent) is the same as running C(docker-compose down).
+    type: str
+    default: present
+    choices:
+      - absent
+      - built
+      - present
+      - pulled
+      - restarted
+      - stopped
+  services:
+    description:
+      - When I(state) is C(present) run C(docker-compose up) resp. C(docker-compose stop) (with I(stopped)) resp. C(docker-compose restart) (with I(restarted))
+        on a subset of services.
+      - If empty, which is the default, the operation will be performed on all services defined in the Compose file (or inline I(definition)).
+    type: list
+    elements: str
+  dependencies:
+    description:
+      - When I(state) is C(present) specify whether or not to include linked services.
+      - When false, equivalent to C(docker-compose up --no-deps).
+      - When I(state) is C(pull) specify whether or not to also pull for linked services.
+      - When true, equivalent to C(docker-compose pull --include-deps).
+    type: bool
+    default: true
+  definition:
+    description:
+      - Compose file describing one or more services, networks and volumes.
+      - Mutually exclusive with I(files).
+    type: dict
+  recreate:
+    description:
+      - By default containers will be recreated when their configuration differs from the service definition.
+      - Setting to C(never) ignores configuration differences and leaves existing containers unchanged.
+      - Setting to C(always) forces recreation of all existing containers.
+      - When set to C(never), equivalent to C(docker-compose up --no-recreate).
+      - When set to C(always), equivalent to C(docker-compose up --force-recreate).
```

Comment:
`smart` is not documented here.

Reply:
It is implied on the line right above (and because the default value for this choice is `smart`):

https://github.com/ansible-collections/community.docker/blob/5d05bedc250356dce166e285bcc6484773329b7e/plugins/modules/docker_compose_v2.py#L112

Should I expand as follow ?:

> By default (when set to C(smart)) containers will be recreated when their configuration differs from the service definition.

I'd prefer to remove the default value for the choice, but I'm not sure this is allowed.

Reply:
I would remove the default. I personally prefer explicit over implicit.

Once the default is gone, the first paragraph should be `When set to C(smart), contains will be recreated ...` or something like that anyway.